core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -373,11 +377,302 @@ User revokeRolesFromUser(String metalake, List<String> roles, String user) {
+                    .withNamespace(roleEntity.namespace())
+                    .withProperties(roleEntity.properties())
+                    .withAuditInfo(auditInfo)
+                    .withSecurableObjects(updateSecurableObjects)
+                    .build();
+              });
+
+      authorizationCallbackWrapper.execute();
+
+      return updatedRole;
+    } catch (NoSuchEntityException nse) {
+      LOG.error("Failed to revoke, role {} does not exist in the metalake {}", role, metalake, nse);
+      throw new NoSuchRoleException(ROLE_DOES_NOT_EXIST_MSG, role, metalake);
+    } catch (IOException ioe) {
+      throw new RuntimeException(ioe);
+    }
+  }
+
+  private static SecurableObject updateRevokedSecurableObject(
+      String metalake,
+      String role,
+      MetadataObject object,
+      List<Privilege> privileges,
+      RoleEntity roleEntity,
+      SecurableObject targetObject,
+      AuthorizationPluginCallbackWrapper authorizationCallbackWrapper) {
+    // If the securable object exists, we remove the privileges of the
+    // securable object. Remove duplicated privileges
+    Set<Privilege> updatePrivileges = Sets.newHashSet();
+    updatePrivileges.addAll(targetObject.privileges());
+    privileges.forEach(updatePrivileges::remove);
+
+    // If the object still contains privilege, we should update the object
+    // with new privileges
+    if (!updatePrivileges.isEmpty()) {
+      SecurableObject newSecurableObject =
+          SecurableObjects.parse(
+              targetObject.fullName(), targetObject.type(), Lists.newArrayList(updatePrivileges));
+      authorizationCallbackWrapper.setCallBack(
+          () ->
+              AuthorizationUtils.callAuthorizationPluginForMetadataObject(
+                  metalake,
+                  object,
+                  authorizationPlugin -> {
+                    authorizationPlugin.onRoleUpdated(
+                        roleEntity,
+                        RoleChange.updateSecurableObject(role, targetObject, newSecurableObject));
+                  }));
+
+      return newSecurableObject;
+    } else {
+      // If the object doesn't contain any privilege, we remove this object.
+      authorizationCallbackWrapper.setCallBack(
+          () ->
+              AuthorizationUtils.callAuthorizationPluginForMetadataObject(
+                  metalake,
+                  object,
+                  authorizationPlugin -> {
+                    authorizationPlugin.onRoleUpdated(
+                        roleEntity, RoleChange.removeSecurableObject(role, targetObject));
+                  }));
+
+      return null;
+    }
+  }
+
+  private List<SecurableObject> updateSecurableObjects(
+      List<SecurableObject> securableObjects,
+      MetadataObject targetObject,
+      Function<SecurableObject, SecurableObject> objectUpdater) {
+    boolean isExist = false;
+    List<SecurableObject> updateSecurableObjects = Lists.newArrayList();
```

Comment:
Also here, can you use a different name?

Reply:
Changed to `RevokedSecurableObjects`.